### PR TITLE
[DNM] Type declarations

### PR DESCRIPTION
**Pull request by kahummer**
The pr adds type definitions to gisida.

### CODE REVIEW

**Comment by cKellyDesign** [comment]
@ciremusyoka @kahummer what's the status of this PR?

**Comment by ciremusyoka** [comment]
> @ciremusyoka @kahummer what's the status of this PR?

@cKellyDesign have realized some changes have been made on the code. Am doing some refactors on the types.

**Comment by cKellyDesign** [comment]
Cool thanks @ciremusyoka

File: src/types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+declare module 'gisida' {
+
+    interface generalFunc {
+        (...args: any[]):any
+    }
+
+    interface generalObjects {
+        [key: string]: any
+    }
```
Comment (cKellyDesign) [inline src/types/index.d.ts]: Can we import/use `FlexObject` definitions from one of the js-tools packages?

File: src/types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
+declare type actionsInterface = import('./actionsInterface').Actions;
+declare type actionTypesInterface = import('./actionTypesInterface').actionTypes;
+
+declare module 'gisida' {
+
+    interface generalFunc {
+        (...args: any[]):any
+    }
```
Comment (cKellyDesign) [inline src/types/index.d.ts]: Nice! Can we call this `FlexFunc` to match `FlexObject` from js-tools?

File: src/store/actions/actions.js [hunk 3]
```diff
@@ -300,12 +295,22 @@ export const loginUser = token => {
...
 
-export const logoutUser = () => dispatch => {
+export const getUserForms = (token) => {
+  const reqConfig = {
+    token,
+    endpoint: 'forms',
+  };
+  // eslint-disable-next-line
+  return dispatch => fetchAPIForms(reqConfig, dispatch);
+};
```
Comment (cKellyDesign) [inline src/store/actions/actions.js]: @ciremusyoka - Why were these actions updated? (7d9436ed4247e138b2ee880144b4a495aeb5fc37)

Comment (ciremusyoka) [reply src/store/actions/actions.js]: @cKellyDesign - Had to rebase with staging to get the latest code version for the type definitions to be up date.

File: src/types/index.d.ts (hunk 1 above)
Comment (ciremusyoka) [reply src/types/index.d.ts]: @cKellyDesign - Have realized the  `FlexObject` and normal `object` have same interface so have just used normal `object`

**Pull request merged**